libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2077,38 +2075,57 @@ string YulUtilFunctions::readFromStorageValueType(Type const& _type, size_t _off
 			_type.identifier();
 
 	return m_functionCollector.createFunction(functionName, [&] {
-		solAssert(_type.sizeOnStack() == 1, "");
-		return Whiskers(R"(
-			function <functionName>(slot) -> value {
-				value := <extract>(sload(slot))
+		Whiskers templ(R"(
+			function <functionName>(slot) -> <?split>addr, selector<!split>value</split> {
+				<?split>let</split> value := <extract>(sload(slot))
+				<?split>
+					addr, selector := <splitFunction>(value)
+				</split>
 			}
-		)")
-				("functionName", functionName)
-				("extract", extractFromStorageValue(_type, _offset, false))
-				.render();
+		)");
+		templ("functionName", functionName);
+		templ("extract", extractFromStorageValue(_type, _offset));
+		FunctionType const* funType = dynamic_cast<FunctionType const*>(&_type);
```

Comment:
Also you can use auto here
```suggestion
		auto const* funType = dynamic_cast<FunctionType const*>(&_type);
```